src/NLog.Extensions.Logging/Config/NLogLoggingConfiguration.cs

```diff
@@ -210,7 +210,7 @@ public LoggingConfigurationElement(IConfigurationSection configurationSection, s
                 var children = _configurationSection.GetChildren();
                 foreach (var child in children)
                 {
-                    if (!child.GetChildren().Any())
+                    if (!child.GetChildren().Any() && child.Value != null)
```

Comment:
Can logic become this:
```c#
                var children = _configurationSection.GetChildren();
                foreach (var child in children)
                {
                    if (!child.GetChildren().Any())
                    {
                        var childValue = child.Value;
                        if (_topElement && IgnoreTopElementChildNullValue(child, childValue))
                            continue;   // Complex object without any properties has no children and null-value (Ex. empty targets-section)

                        yield return new KeyValuePair<string, string>(GetConfigKey(child), childValue);
                    }
                }
```

```c#
            private static bool IgnoreTopElementChildNullValue(IConfigurationSection child, object childValue)
            {
                if (childValue is null)
                {
                    var childName = GetConfigKey(child);
                    if (string.Equals(TargetsKey, childName, StringComparison.OrdinalIgnoreCase)
                      || string.Equals(VariablesKey, childName, StringComparison.OrdinalIgnoreCase)
                      || string.Equals("Rules", childName, StringComparison.OrdinalIgnoreCase)
                      || string.Equals("Extensions", childName, StringComparison.OrdinalIgnoreCase))
                        return true;
                }

                return false;
            }
```

Reply:
Applied the change, updated corresponding tests

Reply:
Looks good